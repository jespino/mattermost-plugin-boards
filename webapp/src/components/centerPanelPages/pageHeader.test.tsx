// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'
import {mocked} from 'jest-mock'
import {Provider as ReduxProvider} from 'react-redux'
import {render} from '@testing-library/react'

import {mockDOM, mockStateStore, wrapDNDIntl} from '../../testUtils'
import {TestBlockFactory} from '../../test/testBlockFactory'
import {IUser} from '../../user'
import octoClient from '../../octoClient'
import {Constants} from '../../constants'

import PageHeader from './pageHeader'
Object.defineProperty(Constants, 'versionString', {value: '1.0.0'})
jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom')

    return {
        ...originalModule,
        useRouteMatch: jest.fn(() => {
            return {url: '/board/view'}
        }),
    }
})
jest.mock('../../octoClient')
jest.mock('../../mutator')
jest.mock('../../telemetry/telemetryClient')
jest.mock('draft-js/lib/generateRandomKey', () => () => '123')
const mockedOctoClient = mocked(octoClient, true)
describe('components/pageHeader', () => {
    const board = TestBlockFactory.createBoard()
    board.id = '1'
    board.teamId = 'team-id'
    const activePage = TestBlockFactory.createPage(board)
    activePage.id = '1'

    const state = {
        clientConfig: {
            value: {
                featureFlags: {
                    subscriptions: true,
                },
            },
        },
        searchText: '',
        users: {
            me: {
                id: 'user_id_1',
            },
            myConfig: {
                onboardingTourStarted: {value: false},
            },
            boardUsers: {
                'user-id-1': {username: 'username_1'},
            },
            blockSubscriptions: [],
        },
        teams: {
            current: {id: 'team-id'},
        },
        boards: {
            current: board.id,
            boards: {
                [board.id]: board,
            },
            templates: [],
            myBoardMemberships: {
                [board.id]: {userId: 'user_id_1', schemeAdmin: true},
            },
        },
        limits: {
            limits: {
                cards: 0,
                used_cards: 0,
                card_limit_timestamp: 0,
                views: 0,
            },
        },
        cards: {
            cards: [],
        },
        views: {
            views: {},
        },
        pages: {
            pages: {[activePage.id]: activePage},
            current: activePage.id,
        },
        contents: {
            contents: [],
            contentsByCard: {},
            contentsByPage: {},
        },
        comments: {
            comments: [],
            commentsByCard: {},
        },
        imits: {
            limits: {
                views: 0,
            },
        },
    }
    mockedOctoClient.searchTeamUsers.mockResolvedValue(Object.values(state.users.boardUsers) as IUser[])
    const store = mockStateStore([], state)
    beforeAll(() => {
        mockDOM()
        console.error = jest.fn()
    })
    beforeEach(() => {
        jest.clearAllMocks()
    })
    test('should match snapshot', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <PageHeader
                    boardId={board.id}
                    activePage={activePage}
                    readonly={false}
                    enablePublicSharedBoards={true}
                    showPage={jest.fn()}
                />
            </ReduxProvider>,
        ))
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot when readonly', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <PageHeader
                    boardId={board.id}
                    activePage={activePage}
                    readonly={true}
                    enablePublicSharedBoards={true}
                    showPage={jest.fn()}
                />
            </ReduxProvider>,
        ))
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot public share boards', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <PageHeader
                    boardId={board.id}
                    activePage={activePage}
                    readonly={false}
                    enablePublicSharedBoards={true}
                    showPage={jest.fn()}
                />
            </ReduxProvider>,
        ))
        expect(container).toMatchSnapshot()
    })
})
