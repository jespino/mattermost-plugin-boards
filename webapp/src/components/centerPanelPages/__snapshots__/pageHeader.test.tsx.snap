// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pageHeader should match snapshot 1`] = `
<div>
  <div
    class="PageHeader"
  >
    <div
      class="mid-head"
    >
      <div
        class="FormattingMenu"
      >
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-1"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-2"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-3"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-4"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-5"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-6"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-bold"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-italic"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-strikethrough-variant"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-link-variant"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-tags"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-block"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-quote-open"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-bulleted"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-numbered"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-table-plus"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-plus"
          />
        </button>
      </div>
      <div
        class="shareButtonWrapper"
      >
        <div
          class="ShareBoardButton"
        >
          <button
            class="Button emphasis--primary size--medium"
            title="Share board"
            type="button"
          >
            <i
              class="CompassIcon icon-lock-outline LockOutlineIcon"
            />
            <span>
              Share
            </span>
          </button>
        </div>
      </div>
      <button
        class="Button size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-message-text-outline"
        />
        <span>
          8
        </span>
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-star-outline"
        />
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-information-outline"
        />
      </button>
      <div
        class="PageMenu"
      >
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="IconButton size--medium"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/pageHeader should match snapshot public share boards 1`] = `
<div>
  <div
    class="PageHeader"
  >
    <div
      class="mid-head"
    >
      <div
        class="FormattingMenu"
      >
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-1"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-2"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-3"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-4"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-5"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-6"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-bold"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-italic"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-strikethrough-variant"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-link-variant"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-tags"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-block"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-quote-open"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-bulleted"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-numbered"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-table-plus"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-plus"
          />
        </button>
      </div>
      <div
        class="shareButtonWrapper"
      >
        <div
          class="ShareBoardButton"
        >
          <button
            class="Button emphasis--primary size--medium"
            title="Share board"
            type="button"
          >
            <i
              class="CompassIcon icon-lock-outline LockOutlineIcon"
            />
            <span>
              Share
            </span>
          </button>
        </div>
      </div>
      <button
        class="Button size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-message-text-outline"
        />
        <span>
          8
        </span>
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-star-outline"
        />
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-information-outline"
        />
      </button>
      <div
        class="PageMenu"
      >
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="IconButton size--medium"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/pageHeader should match snapshot when readonly 1`] = `
<div>
  <div
    class="PageHeader"
  >
    <div
      class="mid-head"
    >
      <div
        class="FormattingMenu"
      >
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-1"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-2"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-3"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-4"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-5"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-header-6"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-bold"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-italic"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-strikethrough-variant"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-link-variant"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-tags"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-code-block"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-quote-open"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-bulleted"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-format-list-numbered"
          />
        </button>
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-table-plus"
          />
        </button>
        <span
          class="divider"
        />
        <button
          class="IconButton size--small"
          type="button"
        >
          <i
            class="CompassIcon icon-plus"
          />
        </button>
      </div>
      <div
        class="shareButtonWrapper"
      >
        <div
          class="ShareBoardLoginButton"
        >
          <button
            class="Button emphasis--primary size--medium"
            title="Login"
            type="button"
          >
            <span>
              Login
            </span>
          </button>
        </div>
      </div>
      <button
        class="Button size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-message-text-outline"
        />
        <span>
          8
        </span>
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-star-outline"
        />
      </button>
      <button
        class="IconButton size--small"
        type="button"
      >
        <i
          class="CompassIcon icon-information-outline"
        />
      </button>
      <div
        class="PageMenu"
      >
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="IconButton size--medium"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
